### PR TITLE
A23

### PR DESCRIPTION
**Pull request by asfar-nazar**

### CODE REVIEW

File: PSICompile/ILCodeGen.cs [hunk 1]
```diff
@@ -66,9 +93,37 @@ public class ILCodeGen : Visitor {
...
+      f.Start.Accept (this);
+      var vd = (NVarDecl)mSymbols.Find (f.Var)!;
+      var type = TMap[vd.Type];
+      StoreVar (f.Var);
+      var cLabel = NextLabel ();
+      var bLabel = NextLabel ();
+      Out ($"    br {cLabel}");
+      Out ($"    {bLabel}:");
+      f.Body.Accept (this);
+      Out ($"    {(vd.Local ? "ldloc " : $"ldsfld {type} Program::")}{vd.Name}");
```
Comment (sdinesh-90) [inline PSICompile/ILCodeGen.cs]: You could write LoadVar similar to StoreVar

File: PSICompile/ILCodeGen.cs [hunk 2]
```diff
@@ -66,9 +93,37 @@ public class ILCodeGen : Visitor {
...
+      Out ("    ldc.i4.1");
+      Out ($"    {(f.Ascending ? "add" : "sub")}");
+      Out ($"    {(vd.Local ? "stloc " : $"stsfld {type} Program::")}{vd.Name}");
```
Comment (sdinesh-90) [inline PSICompile/ILCodeGen.cs]: You could use StoreVar

**Review by sdinesh-90 — CHANGES_REQUESTED** [review]

File: PSICompile/ILCodeGen.cs (hunk 2 above)
Comment (asfar-nazar) [reply PSICompile/ILCodeGen.cs]: Changed

File: PSICompile/ILCodeGen.cs (hunk 1 above)
Comment (asfar-nazar) [reply PSICompile/ILCodeGen.cs]: Added LoadVar helper method

**Comment by sdinesh-90** [comment]
Looks fine.

**Pull request closed without merging**